src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/NuGetWorkspaceCommandHandler.cs

```diff
@@ -65,24 +79,64 @@ public bool QueryStatus(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGr
             if (pguidCmdGroup == CommandGroup.NuGetOnlineEnvironmentsClientProjectCommandSetGuid)
             {
                 var nCmdIDInt = (int)nCmdID;
-
-                if (IsSolutionOnlySelection(selection))
                 {
                     switch (nCmdIDInt)
                     {
                         case PkgCmdIDList.CmdidRestorePackages:
-                            var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
-                            cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
-                                | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
-                            handled = true;
+                            if (IsSolutionOnlySelection(selection))
+                            {
+
+                                var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
+                                cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
+                                    | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
+                                handled = true;
+                            }
+                            break;
+                        case PkgCmdIDList.CmdIdManageProjectUI:
+                            if (IsSupportedProjectSelection(selection))
+                            {
+                                var isPackageManagerUISupported = _packageManagerUICommandHandler.IsPackageManagerUISupported(selection.Single());
+                                cmdf = (uint)(isPackageManagerUISupported ?
+                                    (Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED) :
+                                    Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_INVISIBLE);
+                                handled = true;
+                            }
+                            break;
+                        default:
                             break;
                     }
                 }
             }
-
             return handled;
         }
 
+        private static bool IsSupportedProjectSelection(List<WorkspaceVisualNodeBase> selection)
```

Comment:
If the selection collection is always going to contain one element then I am trying to understand why we have a `List<WorkspaceVisualNodeBase>`.

Reply:
It'll be 1, but the list is something we get from the interface. 
Given that this is a private method, it's easier to just contain the logic within the same method.

Reply:
Keep the logic with the full list also allows us to add any multi selection items if necessary.